[2025-10-11T17:52:38.124+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:52:38.125+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:52:38.167+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:52:38.168+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:52:38.343+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:51:33.907289+00:00 [queued]>
[2025-10-11T17:52:38.353+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:51:33.907289+00:00 [queued]>
[2025-10-11T17:52:38.354+0000] {taskinstance.py:1359} INFO - Starting attempt 2 of 3
[2025-10-11T17:52:38.368+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): bronze_rig_count_pipeline> on 2025-10-11 17:51:33.907289+00:00
[2025-10-11T17:52:38.374+0000] {standard_task_runner.py:57} INFO - Started process 637 to run task
[2025-10-11T17:52:38.377+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ETL_pipeline', 'bronze_rig_count_pipeline', 'manual__2025-10-11T17:51:33.907289+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/ETL_pipeline.py', '--cfg-path', '/tmp/tmpmku56kqt']
[2025-10-11T17:52:38.380+0000] {standard_task_runner.py:85} INFO - Job 11: Subtask bronze_rig_count_pipeline
[2025-10-11T17:52:38.433+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:52:38.433+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:51:33.907289+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:52:38.436+0000] {task_command.py:415} INFO - Running <TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:51:33.907289+00:00 [running]> on host 1096c99751cc
[2025-10-11T17:52:38.505+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='bech' AIRFLOW_CTX_DAG_ID='ETL_pipeline' AIRFLOW_CTX_TASK_ID='bronze_rig_count_pipeline' AIRFLOW_CTX_EXECUTION_DATE='2025-10-11T17:51:33.907289+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-11T17:51:33.907289+00:00'
[2025-10-11T17:52:38.507+0000] {bronze_layer.py:105} INFO - Starting bronze layer data ingestion...
[2025-10-11T17:52:38.507+0000] {logging_mixin.py:151} INFO - Getting rig count data...
[2025-10-11T17:52:39.175+0000] {logging_mixin.py:151} INFO - Failed to download rig count data: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:52:39.175+0000] {bronze_layer.py:140} ERROR - Bronze layer ingestion failed: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:52:39.176+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 142, in import_optional_dependency
    module = importlib.import_module(name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipeline/bronze_layer.py", line 111, in bronze_etl
    raw_data = get_rig_count_data(RIG_COUNT_URL, False)
  File "/opt/airflow/pipeline/bronze_layer.py", line 53, in get_rig_count_data
    df = pd.read_excel(BytesIO(response.content), sheet_name='NAM Weekly',
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 478, in read_excel
    io = ExcelFile(io, storage_options=storage_options, engine=engine)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 1513, in __init__
    self._reader = self._engines[engine](self._io, storage_options=storage_options)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_openpyxl.py", line 548, in __init__
    import_optional_dependency("openpyxl")
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 145, in import_optional_dependency
    raise ImportError(msg)
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:52:39.187+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=ETL_pipeline, task_id=bronze_rig_count_pipeline, execution_date=20251011T175133, start_date=20251011T175238, end_date=20251011T175239
[2025-10-11T17:52:39.197+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 11 for task bronze_rig_count_pipeline (Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.; 637)
[2025-10-11T17:52:39.234+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-10-11T17:52:39.247+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
